signatures/alert_signatures.py

```diff
@@ -0,0 +1,1012 @@
+    "GPL FTP CWD ~ attempt": MicroAttackStage.INFO_DISC,
+    "GPL FTP CWD ~root attempt": MicroAttackStage.INFO_DISC,
+    "GPL FTP LIST directory traversal attempt": MicroAttackStage.INFO_DISC,
+    "GPL FTP MKD overflow": MicroAttackStage.INFO_DISC,
+    "GPL FTP MKD overflow attempt": MicroAttackStage.INFO_DISC,
+    "GPL FTP PORT bounce attempt": MicroAttackStage.INFO_DISC,
+    "GPL FTP SITE EXEC attempt": MicroAttackStage.INFO_DISC,
+    "GPL ICMP_INFO PING *NIX": MicroAttackStage.SERVICE_DISC,
+    "GPL ICMP_INFO PING BSDtype": MicroAttackStage.SERVICE_DISC,
+    "GPL MISC UPnP malformed advertisement": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL MISC rsh root": MicroAttackStage.ROOT_PRIV_ESC,
+    "GPL NETBIOS DCERPC IActivation little endian bind attempt": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL NETBIOS DCERPC Remote Activation bind attempt": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL NETBIOS SMB-DS ADMIN$ share access": MicroAttackStage.DATA_EXFILTRATION,
+    "GPL NETBIOS SMB-DS C$ share access": MicroAttackStage.DATA_EXFILTRATION,
+    "GPL NETBIOS SMB-DS C$ unicode share access": MicroAttackStage.DATA_EXFILTRATION,
+    "GPL NETBIOS SMB-DS D$ share access": MicroAttackStage.DATA_EXFILTRATION,
+    "GPL NETBIOS SMB-DS IPC$ share access": MicroAttackStage.DATA_EXFILTRATION,
+    "GPL NETBIOS SMB-DS IPC$ unicode share access": MicroAttackStage.DATA_EXFILTRATION,
+    "GPL NETBIOS SMB-DS Session Setup NTMLSSP asn1 overflow attempt": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL NETBIOS SMB-DS Session Setup NTMLSSP unicode asn1 overflow attempt": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL NETBIOS SMB-DS repeated logon failure": MicroAttackStage.BRUTE_FORCE_CREDS,
+    "GPL POLICY Sun JavaServer default password login attempt": MicroAttackStage.BRUTE_FORCE_CREDS,
+    "GPL POP3 POP3 PASS overflow attempt": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL RPC portmap bootparam request TCP": MicroAttackStage.SERVICE_DISC ,
+    "GPL RPC portmap cachefsd request TCP": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL RPC portmap listing TCP 111": MicroAttackStage.SERVICE_DISC,
+    "GPL RPC portmap listing UDP 111": MicroAttackStage.SERVICE_DISC,
+    "GPL RPC portmap mountd request UDP": MicroAttackStage.SERVICE_DISC,
+    "GPL RPC portmap rstatd request TCP": MicroAttackStage.SERVICE_DISC,
+    "GPL RPC portmap rusers request TCP": MicroAttackStage.SERVICE_DISC,
+    "GPL RPC portmap sadmind request TCP": MicroAttackStage.SERVICE_DISC,
+    "GPL RPC portmap ypserv request TCP": MicroAttackStage.SERVICE_DISC,
+    "GPL RPC portmap ypupdated request TCP": MicroAttackStage.ARBITRARY_CODE_EXE,
+    "GPL RPC xdmcp info query": MicroAttackStage.INFO_DISC,
+    "GPL SNMP private access udp": MicroAttackStage.ACCT_MANIP,  # Check this out, this is a bit unclear
+    "GPL SNMP public access udp": MicroAttackStage.ACCT_MANIP,
+    "GPL WEB_SERVER globals.pl access": MicroAttackStage.INFO_DISC,  # Changed from exfiltration
```

Comment:
Also comments like this...

Reply:
Yes, since the signature mapping can sometimes be unclear. Good analysis points for whoever works on these mappings later